File: state/Desafio/Camila Aparecida_DesafioState/DesafioState/src/desafiostate/Eduardo Junior - DesafioState/src/MarioFogo.java
public class MarioFogo implements MarioState {

    @Override
    public MarioState pegarCogumelo() {
        System.out.println("Ganha 1000 pontos");
        return this;
    }

    @Override
    public MarioState pegarFlor() {
        System.out.println("Ganha 1000 pontos");
        return this;
    }

    @Override
    public MarioState pegarPena() {
        System.out.println("Mario Fogo");
        return this; 
    }

    @Override
    public MarioState levarDano() {
        System.out.println("Mario Grande");
        return new MarioGrande();
    }   
}

File: state/Desafio/Renny/PadraoState/Entitades/MarioFogo.cs
using PadraoState.Interfaces;

namespace PadraoState.Entitades
{
    class MarioFogo : MarioState
    {
        public MarioFogo()
        {
            System.Console.WriteLine("Mario Fogo");
        }

        public MarioState LevarDano()
        {
            return new MarioGrande();
        }

        public MarioState PegarCogumelo()
        {
            System.Console.Write("Pontos +1000 - ");
            return new MarioFogo();
        }

        public MarioState PegarFlor()
        {
            System.Console.Write("Pontos +1000 - ");
            return new MarioFogo();
        }

        public MarioState PegarPena()
        {
            return new MarioFogo();
        }
    }
}
